integration/test_collections_sorting.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from integration.conftest import ClientFactory
+
+
+@pytest.fixture
+def sorting_test_client(client_factory: ClientFactory):
+    client = client_factory()
+    client.collections.delete_all()
+    yield client
+    client.collections.delete_all()
+    client.close()
```

Comment:
I like the idea of making a specific collection but the `client.collections.delete_all()` calls are likely to cause data race issues with other concurrent tests that run in the CI pipeline. I think is perhaps the origin of the flakes within the CI runs of this PR

In `integration.conftest`, there is the `CollectionFactory`. Can you use that here instead? It handles automatically creating collections for you with unique names and subsequently deleting them when the test finishes. wdyt?